src/frontend/mame/luaengine_input.cpp

```diff
@@ -312,6 +313,8 @@ void lua_engine::initialize_input(sol::table &emu)
 	ioport_field_type["player"] = sol::property(&ioport_field::player, &ioport_field::set_player);
 	ioport_field_type["mask"] = sol::property(&ioport_field::mask);
 	ioport_field_type["defvalue"] = sol::property(&ioport_field::defvalue);
+	ioport_field_type["minvalue"] = sol::property(&ioport_field::minval);
+	ioport_field_type["maxvalue"] = sol::property(&ioport_field::maxval);
```

Comment:
Are these valid for digital fields?  If not, they should probably return nil in that case.

Reply:
Valid, but hard to tell how useful. `minvalue` seems to be 0 and `maxvalue` seems to be `defvalue`, which may be confusing for simple buttons.

Reply:
If `maxvalue` is the same as `defvalue` for buttons, then it would be zero (same as `minvalue`) for active high buttons.  That’s definitely unhelpful.

Reply:
To make them return `nil` for non-analog fields, I have to make them a function instead, like this
```C++
	ioport_field_type["minvalue"] = 
			[this] (ioport_field &f) -> sol::object {
				{
					return f.is_analog() ? sol::make_object(sol(), f.minval()) : sol::lua_nil;
				}
			};
```
Is it the right approach?

Reply:
Does it not work with `std::optional`?  In many cases that’s handled intuitively, like:
```
    ioport_field_type["minval'] = sol::property(
            [] (ioport_field &f)
            {
                return f.is_analog() ? std::make_optional(f.minval()) : std::nullopt;
            });
```

(I haven’t actually tested that code, I’m not sure if the `? :` will correctly deduce the common type.  If it doesn’t, replace `std::nullopt` with `std::optional<ioport_value>()` to give it a hand.)

If it doesn’t work with `std::optional` you could try making it return `sol::optional`.  At any rate, `sol::property` works wit lambdas, and non-capturing lambdas are generally better to use when possible.